<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1417</width>
    <height>687</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>1920</width>
    <height>1080</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Keras Model - Uncertainty Analyzer</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <property name="styleSheet">
   <string notr="true">selection-color: rgb(150, 150, 150);</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>false</bool>
  </property>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>30</y>
     <width>111</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Single Image</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="overlap_box">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>480</y>
     <width>141</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Overlap all of the above</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="noise_checkbox">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>350</y>
     <width>121</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Add Noise to Image</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="noiselevel_box">
   <property name="geometry">
    <rect>
     <x>530</x>
     <y>370</y>
     <width>42</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>255</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>370</y>
     <width>111</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Noise Level (0-255)</string>
   </property>
  </widget>
  <widget class="QComboBox" name="model">
   <property name="geometry">
    <rect>
     <x>890</x>
     <y>490</y>
     <width>69</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>VGG16</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Custom</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>18</x>
     <y>319</y>
     <width>146</width>
     <height>16</height>
    </rect>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="text">
    <string>Setup</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QCheckBox" name="lrp_checkbox">
   <property name="geometry">
    <rect>
     <x>28</x>
     <y>339</y>
     <width>51</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>LRP</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="checkBox_3">
   <property name="geometry">
    <rect>
     <x>28</x>
     <y>379</y>
     <width>77</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>GRAD-CAM</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="monte_carlo_checkbox">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>500</y>
     <width>151</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Monte Carlo Dropout</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="lime_checkbox">
   <property name="geometry">
    <rect>
     <x>28</x>
     <y>429</y>
     <width>51</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>LIME</string>
   </property>
  </widget>
  <widget class="QPushButton" name="button_analyze">
   <property name="geometry">
    <rect>
     <x>1090</x>
     <y>490</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Analyze</string>
   </property>
  </widget>
  <widget class="QPushButton" name="button_load_single_image">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>22</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>Load Image</string>
   </property>
  </widget>
  <widget class="QComboBox" name="lrp_rule_box">
   <property name="geometry">
    <rect>
     <x>68</x>
     <y>358</y>
     <width>131</width>
     <height>21</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>lrp.alpha_1_beta_0</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string/>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>32</x>
     <y>359</y>
     <width>31</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Rule</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>16</x>
     <y>449</y>
     <width>61</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Samples:</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QSpinBox" name="lime_samples_box">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>450</y>
     <width>47</width>
     <height>21</height>
    </rect>
   </property>
   <property name="minimum">
    <number>50</number>
   </property>
   <property name="maximum">
    <number>5000</number>
   </property>
   <property name="singleStep">
    <number>50</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>520</y>
     <width>61</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Samples</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QSpinBox" name="spinBox_2">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>520</y>
     <width>41</width>
     <height>18</height>
    </rect>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QLabel" name="single_image_label">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>70</y>
     <width>224</width>
     <height>224</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: white</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QCheckBox" name="gradcam_checkbox">
   <property name="geometry">
    <rect>
     <x>28</x>
     <y>379</y>
     <width>91</width>
     <height>17</height>
    </rect>
   </property>
   <property name="mouseTracking">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>GRAD-CAM</string>
   </property>
   <property name="checkable">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>460</x>
     <y>320</y>
     <width>47</width>
     <height>13</height>
    </rect>
   </property>
   <property name="text">
    <string>Noise</string>
   </property>
  </widget>
  <widget class="QComboBox" name="analyze_mode">
   <property name="geometry">
    <rect>
     <x>980</x>
     <y>490</y>
     <width>91</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>Single Image</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Many Images</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Video</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>38</y>
     <width>71</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Many Images</string>
   </property>
  </widget>
  <widget class="QListWidget" name="image_list_widget">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>80</y>
     <width>201</width>
     <height>192</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_10">
   <property name="geometry">
    <rect>
     <x>930</x>
     <y>40</y>
     <width>161</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>GRAD CAM Video Analyzer</string>
   </property>
  </widget>
  <widget class="QPushButton" name="button_load_many_images">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>30</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>Load Image</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="time_series_checkbox">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>40</y>
     <width>161</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Images are time series</string>
   </property>
  </widget>
  <widget class="QPushButton" name="button_load_video">
   <property name="geometry">
    <rect>
     <x>1090</x>
     <y>40</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Load Video</string>
   </property>
  </widget>
  <widget class="QTextBrowser" name="video_text">
   <property name="geometry">
    <rect>
     <x>930</x>
     <y>70</y>
     <width>281</width>
     <height>41</height>
    </rect>
   </property>
  </widget>
  <widget class="QCheckBox" name="noise_walk_checkbox">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>400</y>
     <width>211</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Noise Walk (Only Single Image)</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="noise_walk_steps">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>440</y>
     <width>42</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>3</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="noise_walk_max">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>440</y>
     <width>42</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>255</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>420</y>
     <width>47</width>
     <height>13</height>
    </rect>
   </property>
   <property name="text">
    <string>Max</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>530</x>
     <y>420</y>
     <width>47</width>
     <height>13</height>
    </rect>
   </property>
   <property name="text">
    <string>Steps</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>138</x>
     <y>449</y>
     <width>61</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Features:</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QSpinBox" name="lime_features_box">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>450</y>
     <width>47</width>
     <height>21</height>
    </rect>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
   <property name="singleStep">
    <number>50</number>
   </property>
  </widget>
  <widget class="QCheckBox" name="heatmap_box_cam">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>380</y>
     <width>141</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Heatmap: CAM</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="heatmap_box_lime">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>430</y>
     <width>141</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Heatmap: LIME</string>
   </property>
  </widget>
  <widget class="QLabel" name="many_images_label">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>80</y>
     <width>224</width>
     <height>224</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: white</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QComboBox" name="grad_cam_version_combobox">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>400</y>
     <width>111</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>GradCam</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>GradCam++</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>400</y>
     <width>47</width>
     <height>13</height>
    </rect>
   </property>
   <property name="text">
    <string>Version</string>
   </property>
  </widget>
  <widget class="QTextBrowser" name="textBrowser">
   <property name="geometry">
    <rect>
     <x>860</x>
     <y>430</y>
     <width>231</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QTextBrowser" name="textBrowser_2">
   <property name="geometry">
    <rect>
     <x>1100</x>
     <y>430</y>
     <width>221</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_14">
   <property name="geometry">
    <rect>
     <x>1050</x>
     <y>370</y>
     <width>91</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Custom Model</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>920</x>
     <y>400</y>
     <width>91</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Load Model</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_2">
   <property name="geometry">
    <rect>
     <x>1160</x>
     <y>400</y>
     <width>91</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Load Weights</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
